tests.py

```diff
@@ -452,6 +452,35 @@ def test_attribute_lookup(self):
         template = Template([data])
         self.assertEqual(template.name, data)
 
+    def test_template_from_json_no_error(self):
+        expected = {
+            'template': {
+                'data': [
+                    {'name': 'name', 'value': 'value'},
+                    {'name': 'name1', 'value': 'value1'},
+                ]
+            }
+        }
+        data = json.dumps(expected)
+        template = Template.from_json(data)
+        self.assertEqual(template.to_dict(), expected)
+
+    def test_template_from_json_collection(self):
+        expected = {
+            'collection': {
+                'template': {
+                    'data': [
+                        {'name': 'name', 'value': 'value'},
+                        {'name': 'name1', 'value': 'value1'},
+                    ]
+                }
+            }
+        }
+        data = json.dumps(expected)
+        with self.assertRaisesRegexp(ValueError, "Not valid Collection\+JSON template data."):
```

Comment:
The only reason to check the message is to make sure it didn't throw an error from elsewhere lower in the bowels of the parsing logic -- trying to be thorough.